javascript/ql/src/RegExp/IdentityReplacement.ql

```diff
@@ -0,0 +1,44 @@
+/**
+ * @name Replacement of a substring with itself
+ * @description Replacing a substring with itself has no effect and may indicate a mistake.
+ * @kind problem
+ * @problem.severity warning
+ * @id js/identity-replacement
+ * @tags correctness
```

Comment:
Have you considered the `security` tag?

Reply:
It's a bit of a stretch, but I suppose it does cover CWE-116, so might as well claim it.